packages/core/src/new-api/internal/new-execution/abi.ts

```diff
@@ -176,22 +176,25 @@ export function validateArtifactEventArgumentParams(
 
   const eventFragment = getEventFragment(iface, eventName);
 
-  if (typeof argument === "string") {
-    for (const input of eventFragment.inputs) {
-      if (input.name === argument) {
-        return;
-      }
-    }
-
-    throw new IgnitionValidationError(
-      `Event ${eventName} of contract ${emitterArtifact.contractName} has no argument named ${argument}`
-    );
-  }
+  const paramType = getEventArgumentParamType(
+    emitterArtifact.contractName,
+    eventName,
+    eventFragment,
+    argument
+  );
 
-  if (eventFragment.inputs.length <= argument) {
-    throw new IgnitionValidationError(
-      `Event ${eventName} of contract ${emitterArtifact.contractName} has only ${eventFragment.inputs.length} arguments, but argument ${argument} was requested`
-    );
+  if (paramType.indexed === true) {
+    // We can't access the value of indexed arguments with dynamic size
+    if (
+      paramType.isArray() ||
+      paramType.isTuple() ||
+      paramType.type === "bytes" ||
+      paramType.type === "string"
+    ) {
```

Comment:
Are there any fixed sized types that also don't get indexed?

Reply:
As discussed on a call: no, because fixed-sized types have 32 bytes (i.e. an EVM word) or less.